cpu/sam0_common/Makefile.include

```diff
@@ -4,24 +4,27 @@ CFLAGS += -DCPU_FAM_$(call uppercase_and_underscore,$(CPU_FAM))
 # Generate ASF compatible model definition
 CFLAGS += -D__$(call uppercase_and_underscore,$(CPU_MODEL))__
 
-# Set ROM and RAM lengths according to CPU model
-ifneq (,$(filter samd21g18a samd21j18a saml21j18b saml21j18a samr21e18a \
-        samr21g18a samr30g18a samr34j18b,$(CPU_MODEL)))
-
-  ROM_LEN ?= 0x40000
-  RAM_LEN ?= 0x8000
-endif
-ifneq (,$(filter saml10e16a saml11e16a,$(CPU_MODEL)))
-  ROM_LEN ?= 64K
-  RAM_LEN ?= 16K
+# Compute CPU_LINE
+LINE   := $(shell echo $(CPU_MODEL) | sed -E -e 's/^sam([a-z][0-9][0-9])(.)([0-9][0-9])(.)/\1 \2 \3 \4/')
+FAMILY := $(word 1, $(LINE))
+TYPE1  := $(word 2, $(LINE))
+MEMORY := $(word 3, $(LINE))
+TYPE2  := $(word 4, $(LINE))
+
+# ROM length is directly encoded in the part number
+ROM_LEN := $(shell echo $$((1 << $(MEMORY))))
+
+# On SAMD21, SAML21 RAM is always ⅛ of ROM
+ifneq (,$(filter samd21 saml21,$(CPU_FAM)))
+  RAM_LEN ?= $(shell echo $$(($(ROM_LEN)/8)))
 endif
-ifneq (,$(filter samd21j17d,$(CPU_MODEL)))
-  ROM_LEN ?= 128K
-  RAM_LEN ?= 16K
+# On SAML1x RAM is always ¼ of ROM
```

Comment:
Not true for SAML11x14. SRAM is 1/2. Otherwise it's true.

Reply:
Is there a list of all SAM L10/L11 parts somewhere? The *SAM L10/L11 Family Data Sheet* strangely omits this table.

Reply:
Chapter 10.1 Embedded Memories
[Page 36](http://ww1.microchip.com/downloads/en/DeviceDoc/SAM-L10L11%20Family-DataSheet%20-%20DS60001513B.pdf)